urllibRequests.py

```diff
@@ -1,12 +1,53 @@
-import urllib.request
-import urllib.parse
+import sys
+if sys.hexversion < 50331648:
+    import urllib2 as uReq
+    import urllib as uPar
+else:
+    import urllib.request as uReq
+    import urllib.parse as uPar
 
+class Response:
+    def __init__(self,data,charset='UTF-8'):
+        self.text = data.decode(charset)
 
-def get(urlStr, params={}):
-    reqdata = urllib.request.Request(urlStr)
+def getEncoding(ctstr):
+    return ctstr[ctstr.find('charset')+8:]
+
+def gunzip(data):
+    if sys.hexversion < 50331648:
+        import zlib
+        return zlib.decompress(data,16+zlib.MAX_WBITS)
+    else:
+        import gzip
+        return gzip.decompress(data)
+
+def inflate(data):
+    import zlib
+    try:
+        return zlib.decompress(data)
+    except zlib.error:
+        return zlib.decompress(data,-zlib.MAX_WBITS)
+
+def get(urlStr,params={}):
+    reqdata = uReq.Request(urlStr)
     reqdata.add_header('User-Agent',
                        'VocabTool/0.2 (https://github.com/RihanWu/vocabtool)')
+    reqdata.add_header('Accept-Encoding','gzip, deflate')
     if params != {}:
-        reqdata.data = urllib.parse.urlencode(params).encode('ascii')
-    req = urllib.request.urlopen(reqdata)
-    return req.read()
+        reqdata.data = uPar.urlencode(params).encode('ascii')
```

Comment:
It seems that this is meant to be a POST request? Should we implement POST request along with GET request?